### PR TITLE
40 create a dockerfile and ci

### PR DESCRIPTION
**Pull request by Pipazoul**
Docker compose file and github action to build frontend and backend

### CODE REVIEW

File: .github/workflows/build-backend.yml
Comment (lutangar) [inline .github/workflows/build-backend.yml]:
```suggestion
      - 'main'
```

File: .github/workflows/build-backend.yml
Comment (lutangar) [inline .github/workflows/build-backend.yml:32]: Out of curiosity... Erasme doesn't have its own registry ?

File: .github/workflows/build-frontend.yml
Comment (lutangar) [inline .github/workflows/build-frontend.yml]:
```suggestion
      - 'main'
```

**Review by lutangar — CHANGES_REQUESTED** [review]

**Review by OlivPi — APPROVED** [review]

File: .github/workflows/build-backend.yml
Comment (racine-p-a) [reply .github/workflows/build-backend.yml:32]: Not for Github. On Gitlab, we use the metropole's registry.

**Review by racine-p-a — APPROVED** [review]

File: .github/workflows/build-backend.yml
Comment (Pipazoul) [reply .github/workflows/build-backend.yml:32]: in this situation it seemed more appropriate to publish the images on a public register, in order to allow the deployment by third parties

**Review by Pipazoul — COMMENTED** [review]
All changes have been reviewed

**Comment by Pipazoul** [comment]
@lutangar  the lint test fails with 
```bash
Run npx nx format:check --base=$BASE_BRANCH
fatal: Not a valid object name dev
fatal: No such ref: 'dev'
dev.docker-compose.yml
Error: Process completed with exit code 1.

```


this pull request is not dev branch, is there a workaround ?

Also all changes have been made i think, can you approve this pr ?

**Comment by lutangar** [comment]
> ```shell
> Run npx nx format:check --base=$BASE_BRANCH
> fatal: Not a valid object name dev
> fatal: No such ref: 'dev'
> dev.docker-compose.yml
> Error: Process completed with exit code 1.
> ```

Actually the "branch" part of the error message isn't related to the actual issue, the relevant part is :
```
dev.docker-compose.yml
```
Your formatting issue could probably be fixed by running `npx nx format:fix` locally.

**Comment by Pipazoul** [comment]
> > ```shell
> > Run npx nx format:check --base=$BASE_BRANCH
> > fatal: Not a valid object name dev
> > fatal: No such ref: 'dev'
> > dev.docker-compose.yml
> > Error: Process completed with exit code 1.
> > ```
> 
> Actually the "branch" part of the error message isn't related to the actual issue, the relevant part is :
> 
> ```
> dev.docker-compose.yml
> ```
> 
> Your formatting issue could probably be fixed by running `npx nx format:fix` locally.

Thanks ! 

Waiting for your approval ;)

**Comment by lutangar** [comment]
Thanks @Pipazoul !

**Pull request merged**
